src/exporter/DaemonMetricCollector.cc

```diff
@@ -306,13 +306,19 @@ DaemonMetricCollector::get_labels_and_metric_name(std::string daemon_name,
     labels["instance_id"] = quote(tmp);
   }
   else if (daemon_name.find("rgw") != std::string::npos) {
-    // fetch intance_id for e.g. "okbvtv" from daemon_name=rgw.foo.ceph-node-00.okbvtv 
-    size_t pos = daemon_name.find_last_of(".");
-    std::string instance_id = "";
-    if (pos != std::string::npos) {
-       instance_id = daemon_name.substr(pos+1);
+    // fetch intance_id for e.g. "hrgsea" from daemon_name=rgw.foo.ceph-node-00.hrgsea.2.94739968030880
+    std::vector<std::string> elems;
+    std::stringstream ss;
+    ss.str(daemon_name);
+    std::string item;
+    while (std::getline(ss, item, '.')) {
+        elems.push_back(item);
+    }
+    if (elems.size() >= 4) {
+      labels["instance_id"] = quote(elems[3]);
+    } else {
+      derr << "Error: Not enough elements in `elems` to parse instance_id from daemon_name=" << daemon_name << dendl;
```

Comment:
Could it be useful to add the item when not located?
`
labels["instance_id"] = 'unknown'; `

Reply:
> Could it be useful to add the item when not located? `labels["instance_id"] = 'unknown';`

We shouldn't add any labels if it doesn;t exist in prometheus.

Reply:
And btw PR is still in draft